code/josh/python/josh_lab_13.py

```diff
@@ -0,0 +1,139 @@
+# Lab 13: Compass
+
+
+class Compass:
+    def __init__(self, heading) -> None:
+        self.heading = heading
+
+# Returns 'N', 'NE', 'E', 'SE', 'S', 'SW', 'W', or 'NW' depending upon range of nearest heading
+    def get_direction(self) -> str:
+        '''Find the nearest cardinal direction of the current heading'''
+        direction = self.heading
+        if direction <= 22:
+            direction = 0
+        elif direction > 22 and direction <= 67:
+            direction =  45
+        elif direction > 67 and direction <= 112:
+            direction = 90
+        elif direction > 112 and direction <= 157:
+            direction =  135
+        elif direction > 157 and direction <= 202:
+            direction =  180
+        elif direction > 202 and direction <= 247:
+            direction =  225
+        elif direction > 247 and direction <= 292:
+            direction =  270
+        elif direction > 292 and direction <= 337:
+            direction =  315
+        elif direction > 337 and direction <= 359:
+            direction =  0
+
+        match direction:
+            case 0:
+                return 'N'
+            case 45:
+                return 'NE'
+            case 90:
+                return 'E'
+            case 135:
+                return 'SE'
+            case 180:
+                return 'S'
+            case 225:
+                return 'SW'
+            case 270:
+                return 'W'
+            case 315:
+                return 'NW'
+
+# Turns 'Compass' left or right based upon user input of turn direction and amount of turn
+    def turn(self, degrees, direction) -> None:
+        self.degrees = degrees
+        self.direction = direction
```

Comment:
These lines aren't doing anything, the attributes are already defined by the init (and direction isn't one of them)